packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -363,6 +364,17 @@ export class InterfaceVpcEndpoint extends VpcEndpoint implements IInterfaceVpcEn
     const securityGroup = new SecurityGroup(this, 'SecurityGroup', {
       vpc: props.vpc
     });
+
+    if (props.vpc.vpcCidrBlock) {
```

Comment:
I do not feel comfortable with doing this outright. I would feel better if it's behind a flag (suggested name: `allowAllVpcTraffic: true|false`).

It would be nicer to implement as:

```ts
this.connections.allowDefaultPortFrom(Peer.ipv4(props.vpc.vpcCidrBlock));
```

But even doing this only works for securitygroups that have an outbound rule. If not, you would still need to do the following:

```ts
endpoint.connections.allowDefaultPortFrom(myAsg);
```

So is this even worth it?